model_analyzer/result/run_config_measurement.py

```diff
@@ -380,11 +405,46 @@ def __lt__(self, other):
 
         return not self.is_better_than(other)
 
-    def _compare_measurements(self, other):
+    def is_passing_constraints(self) -> bool:
+        """
+        Returns true if all model measurements pass
+        their respective constraints
+        """
+        return ConstraintManager.satisfies_constraints(
+            self._model_config_constraints, self)
+
+    def compare_measurements(self, other: 'RunConfigMeasurement') -> float:
```

Comment:
I think [PEP 484](https://peps.python.org/pep-0484/#forward-references) recommends using string literals only when there's cyclic import issue or the class has not been defined yet. But since I'm also pretty new to type hints, this may not be a big issue. Any thoughts?

Reply:
This is the exact case they used in your link -- a function in class RunConfigMeasurement that takes a RunConfigMeasurement as an argument.

Reply:
Oh! Haha totally forgot that this was inside RunConfigMeasurement 😅